pagefaultgames/pokerogue
# [Bug] Fixed primal weather remaining active during transition from double to single battle

## What are the changes?
Fixes a bug where primal weather would not go away when the mon with the ability is forced back into it's ball during the transition from a double to a single battle.

## Why am I doing these changes?
Relevant issue: https://github.com/pagefaultgames/pokerogue/issues/3385

## What did change?


The `ReturnPhase` class that controls the forced switch going from a double to a single battle now calls the `applyPreSwitchOutAbAttrs` function for controlling how abilities behave during a switch.

### Screenshots/Videos



## How to test the changes?




Used the session file from the relevant issue until I got to a double battle and observed that Desolate Land correctly got deactivated.

Unsure if this change would justify any automated tests so if I should add some please let me know!

## Checklist
